libchisel/src/remapstart.rs

```diff
@@ -0,0 +1,259 @@
+use parity_wasm::elements::*;
+use std::collections::HashMap;
+
+use super::{ModuleError, ModulePreset, ModuleTranslator};
+use super::utils::HasNamesSection;
+
+pub struct RemapStart;
+
+impl ModulePreset for RemapStart {
+    fn with_preset(preset: &str) -> Result<Self, ()> {
+        match preset {
+            // TODO refactor this later
+            "ewasm" => Ok(RemapStart {}),
+            _ => Err(()),
+        }
+    }
+}
+
+impl ModuleTranslator for RemapStart {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Ok(remap_start(module))
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let mut ret = module.clone();
+        if remap_start(&mut ret) {
+            Ok(Some(ret))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+fn clear_names_section(module: &mut Module) {
+    let sections = module.sections_mut();
+    let mut rmidx = sections.len();
+    for (index, section) in sections.iter_mut().enumerate() {
```

Comment:
This can be pulled out into `find_names_section(module) -> Option(usize)` which then simplifies this code – no need for the `if` below.